ess/src/API/EMBC.ESS/Resources/Suppliers/SupplierRepository.cs

```diff
@@ -52,15 +52,44 @@ private async Task<SupplierQueryResult> HandleQuery(SupplierQuery queryRequest)
             IQueryable<era_essteamsupplier> supplierQuery = essContext.era_essteamsuppliers
                 .Expand(s => s.era_SupplierId)
                 .Expand(s => s.era_ESSTeamID)
-                .Where(s => s.statecode == (int)EntityState.Active)
-                .Where(s => s._era_essteamid_value == Guid.Parse(queryRequest.TeamId));
+                .Where(s => s.statecode == (int)EntityState.Active);
+
+            if (!string.IsNullOrEmpty(queryRequest.TeamId)) supplierQuery = supplierQuery.Where(s => s._era_essteamid_value == Guid.Parse(queryRequest.TeamId));
+            if (!string.IsNullOrEmpty(queryRequest.SupplierId)) supplierQuery = supplierQuery.Where(s => s._era_supplierid_value == Guid.Parse(queryRequest.SupplierId));
 
             var teamSuppliers = await ((DataServiceQuery<era_essteamsupplier>)supplierQuery).GetAllPagesAsync();
 
+            if (!string.IsNullOrEmpty(queryRequest.LegalName) && !string.IsNullOrEmpty(queryRequest.GSTNumber)) teamSuppliers = teamSuppliers.Where(s => s.era_SupplierId.era_supplierlegalname == queryRequest.LegalName && s.era_SupplierId.era_gstnumber == queryRequest.GSTNumber);
+
             var items = mapper.Map<IEnumerable<Supplier>>(teamSuppliers);
 
             foreach (var ts in items)
             {
+                if (ts.IsPrimarySupplier)
```

Comment:
this entire block can be condensed to a loadproperty of the teamsupplier collection and LINQ to map each property